src/sinks/datadog/logs/sink.rs

```diff
@@ -19,12 +19,37 @@ use crate::{
 #[derive(Default)]
 pub struct EventPartitioner;
 
+impl EventPartitioner {
+    /// Evaluates the payload as conforming to the agent standard if the following 7 attributes
+    /// exist and also adhere to the expected types. All types are expected to be strings with the
+    /// exception of timestamp which should be an integer.
+    fn event_conforms_to_agent_payload(item: &Event) -> bool {
+        let x = true;
+        let attr_is =
+            |log: &LogEvent, attr: &str, f: fn(&Value) -> bool| log.get(attr).map(f).is_some();
+        let attr_is_bytes = |log, attr| attr_is(log, attr, |f| f.is_bytes());
+        let attr_is_int = |log, attr| attr_is(log, attr, |f| f.is_integer());
```

Comment:
These closures don't actually capture anything, so it would be clearer to write them as plain functions.

Also `|f: &Value| f.is_bytes()` is `Value::is_bytes` etc